### DISCUSSION TITLE
Configuration of available basemaps

### DISCUSSION THREAD

**Discussion by alexgleith** [TerriaJS/terriajs#6922] [General] [answered]
Hey folks!

I'm doing a lazy build of TerriaMap [over here](https://github.com/digitalearthpacific/dep-maps), based on your existing
Docker image and I can't work out how to change what basemaps are available.

Is it possible to change what basemaps are available using the standard build of TerriaMap? Or do I need to get
a custom build going?

Thanks!

**Comment by zoran995** [comment, marked as answer]
Basemaps can be configured via initialization files, you can find doc here https://docs.terria.io/guide/customizing/initialization-files/#basemaps. There is a https://github.com/TerriaJS/terriajs/pull/6924 to fix the formatting of doc table.

**Reply by alexgleith** [reply]
Oh, great! So it's done on the layer config... missed that somehow.

Can we _remove_ the default basemaps?

**Reply by alexgleith** [reply]
Oh I see `enabledBasemaps`

**Reply by nf-s** [reply]
Yep!

Here is an example configuration - for Digital Earth Australia map

```json
{
  "baseMaps": {
    "defaultBaseMapId": "basemap-bing-aerial-with-labels",
    "previewBaseMapId": "basemap-positron",
    "enabledBaseMaps": [
      "basemap-australian-topography",
      "basemap-bing-aerial",
      "basemap-bing-aerial-with-labels",
      "basemap-bing-roads",
      "basemap-natural-earth-II",
      "basemap-black-marble",
      "basemap-positron",
      "basemap-darkmatter"
    ],
    "items": [
      {
        "item": {
          "id": "basemap-australian-topography",
          "type": "esri-mapServer",
          "name": "Australian Topography",
          "url": "https://services.ga.gov.au/gis/rest/services/NationalBaseMap/MapServer",
          "opacity": 1.0
        },
        "image": "images/basemaps/australian-topo.png"
      }
    ]
  }
}
```
